src/servers/src/http/influxdb_result_v1.rs

```diff
@@ -0,0 +1,229 @@
+                        .into_iter()
+                        .map(|value| {
+                            if let Some(epoch) = epoch {
+                                if let datatypes::value::Value::Timestamp(ts) = &value {
+                                    if let Some(timestamp) = epoch.convert_timestamp(*ts) {
+                                        return datatypes::value::Value::Timestamp(timestamp);
+                                    }
+                                }
+                            }
+
+                            value
+                        })
+                        .map(Value::try_from)
+                        .collect::<Result<Vec<Value>, _>>()
+                        .context(ToJsonSnafu)?;
+
+                    rows.push(value_row);
+                }
+            }
+
+            Ok(InfluxdbRecordsOutput::new(Some(columns), rows))
+        }
+    }
+}
+
+#[derive(Serialize, Deserialize, Debug, JsonSchema)]
+pub struct InfluxdbOutput {
+    pub statement_id: u32,
+    pub series: Vec<InfluxdbRecordsOutput>,
+}
+
+#[derive(Serialize, Deserialize, Debug, JsonSchema)]
+pub struct InfluxdbV1Response {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    results: Option<Vec<InfluxdbOutput>>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    error: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    execution_time_ms: Option<u128>,
+}
+
+impl InfluxdbV1Response {
+    pub fn with_error(error: impl ErrorExt) -> Self {
+        let code = error.status_code();
+        if code.should_log_error() {
+            error!(error; "Failed to handle HTTP request");
+        } else {
+            debug!("Failed to handle HTTP request, err: {:?}", error);
+        }
+
+        InfluxdbV1Response {
+            results: None,
+            error: Some(error.output_msg()),
+            execution_time_ms: None,
+        }
+    }
+
+    pub fn with_error_message(err_msg: String) -> Self {
+        InfluxdbV1Response {
+            results: None,
+            error: Some(err_msg),
+            execution_time_ms: None,
+        }
+    }
+
+    fn with_output(results: Option<Vec<InfluxdbOutput>>) -> Self {
+        InfluxdbV1Response {
+            results,
+            error: None,
+            execution_time_ms: None,
+        }
+    }
+
+    pub fn with_execution_time(&mut self, execution_time: u128) {
+        self.execution_time_ms = Some(execution_time);
+    }
+
+    /// Create a influxdb v1 response from query result
+    pub async fn from_output(
+        outputs: Vec<crate::error::Result<Output>>,
+        epoch: Option<Epoch>,
+    ) -> Self {
+        // TODO(sunng87): this api response structure cannot represent error
+        // well. It hides successful execution results from error response
+        let mut results = Vec::with_capacity(outputs.len());
+        for (statement_id, out) in outputs.into_iter().enumerate().map(|(i, o)| (i as u32, o)) {
```

Comment:
We can move this `map` stage into `for` iterations.
```suggestion
        for (statement_id, out) in outputs.into_iter().enumerate() {
            let i = i as u32;
```